lib/diagnostic.js

```diff
@@ -6,6 +6,13 @@
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
 const min = function (/* previous, current */) {
+  let minimum = arguments[0];
+
+  for (let i = 0; i < arguments.minimum; i++) {
```

Comment:
Both arguments to ```min()``` are assumed to be numbers.  ```arguments``` doesn't have a property called ```minimum``` so ```arguments.minimum``` evaluates to ```undefined```.